cow_instrument/InstrumentTree.cpp

```diff
@@ -18,53 +18,68 @@ void findDetectors(const Component &component,
 }
 
 /*
- * Derived, detector vector is always recreated.
+ * Derived, detector vector is always recreated. Except on a plain-old copy.
 */
-InstrumentTree::InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors)
-    : m_nodes(std::move(nodes)) {
-
+void InstrumentTree::init(size_t nDetectors) {
   // This will make the push_backs faster
-  m_detectorVec.reserve(nDetectors);
+  m_detectorVec->reserve(nDetectors);
```

Comment:
Why do you expect the reserve call to have a negligible effect? It should put the `vector` at the correct capacity. Are you proposing removing the `reserve`?